core/ibft.go

```diff
@@ -544,6 +544,9 @@ func (i *IBFT) runNewRound(ctx context.Context) error {
 				MessageType:    proto.MessageType_PREPREPARE,
 				View:           view,
 				MinNumMessages: 1,
+				HasQuorumFn: func(view *proto.View, messages []*proto.Message) bool {
```

Comment:
why do we need to repeat this code? could we have a function?

Reply:
of course

Reply:
These are special cases also present in the original implementation, where the default behaviour of the `HasQuorum` backend function differs for the subscribe.